subsys/usb/class/cdc_acm.c

```diff
@@ -243,6 +243,8 @@ static void tx_work_handler(struct k_work *work)
 		return;
 	}
 
+	dev_data->tx_ready = false;
```

Comment:
>                        - ringbuffer data are finished
>                          Now, we are in a dangerous situation
>                          because tx_ready is on true state and data
>                          released in ringbuffer are being transmitted

It does not explain why it is "dangerous".
[For the trivial case of one producer and one consumer, concurrency control shouldn’t be needed.](https://docs.zephyrproject.org/latest/reference/data_structures/ring_buffers.html?highlight=ring_buffer#concurrency)
Anyway NACK for commit message.

Reply:
In the function tx_work_handler, a pointer to ring buffer data is given to usb_transfer then the ring_buf_get_finish is called. So, the data is mark as read (by ring_buf_get_finish) while the data are not yet transferred to usb. If later a user send data, the pointer hold by usb stack could be rewrite.

Reply:
> In the function tx_work_handler, a pointer to ring buffer data is given to usb_transfer then the ring_buf_get_finish is called. So, the data is mark as read (by ring_buf_get_finish) while the data are not yet transferred to usb. If later a user send data, the pointer hold by usb stack could be rewrite.

This :arrow_up: should be the commit message, please update.

Here it is assumed that with usb_transfer() (usb_dc_ep_write)  the data is copied into an endpoint buffer/FIFO, but yes I agree this does not have to be always the case. Next, I think it would be better to use atomic_t for tx_ready/rx_ready/tx_irq_ena/rx_irq_ena.